onelogin/get.go

```diff
@@ -107,36 +121,85 @@ func Get(app, provider string) (*awsprovider.Credentials, error) {
 		fmt.Scanln(&selection)
 
 		deviceID = fmt.Sprintf("%v", devices[selection-1].DeviceId)
+		deviceType = devices[selection-1].DeviceType
+
 	} else {
 		deviceID = fmt.Sprintf("%v", devices[0].DeviceId)
+		deviceType = devices[0].DeviceType
 	}
 
-	fmt.Print("Please enter the OTP from your MFA device: ")
-	var otp string
-	fmt.Scanln(&otp)
+	var rMfa *VerifyFactorResponse
 
-	// Verify MFA
-	pMfa := VerifyFactorParams{
-		AppId:      a.ID,
-		DeviceId:   deviceID,
-		StateToken: st,
-		OtpToken:   otp,
-	}
+	var pushOK = false
 
-	s.Start()
-	rMfa, err := c.VerifyFactor(token, &pMfa)
-	s.Stop()
-	if err != nil {
-		return nil, fmt.Errorf("verifying factor: %v", err)
+	if deviceType == MFADeviceOneLoginProtect {
+		// push is supported by the selected OTP device
+
+		// let's assume push will succeed
+		pushOK = true
+		s.Start()
+		pMfa := VerifyFactorParams{
+			AppId:       a.ID,
+			DeviceId:    deviceID,
+			StateToken:  st,
+			OtpToken:    "",
+			DoNotNotify: false,
+		}
+
+		rMfa, err = c.VerifyFactor(token, &pMfa)
+		if err != nil {
+			return nil, err
+		}
+
+		pMfa.DoNotNotify = true
+
+		s.Stop()
+		fmt.Println(rMfa.Status.Message)
+		s.Start()
```

Comment:
This should be right before we start sleeping in a loop. At the moment there is a var assignment in the middle...